src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/MonitorConnectionContext.java

```diff
@@ -200,7 +200,7 @@ void setConnectionValid(
 
       final long invalidNodeDurationMillis = currentTime - this.getInvalidNodeStartTime();
       final long maxInvalidNodeDurationMillis =
-          (long) this.getFailureDetectionIntervalMillis() * this.getFailureDetectionCount();
+          (long) this.getFailureDetectionIntervalMillis() * Math.max(0, this.getFailureDetectionCount() - 1);
```

Comment:
NIT: Can this counter go below 0?

Reply:
It shouldn't but it can. That's why I use 0 in the worst case scenario.

Reply:
Cool. Just to be sure. Also because you're multiplying it later, so multiplying by 0 is something often unwanted